### PR TITLE
Do not enable APM agent 'instrument', it's not required for manual tracing.

### PR DESCRIPTION
**Pull request by mosche**
Do not enable APM agent 'instrument', it's not required for manual tracing.

### CODE REVIEW

File: modules/apm/src/main/java/org/elasticsearch/telemetry/apm/internal/APMAgentSettings.java [hunk 1]
```diff
@@ -120,7 +118,8 @@ public void setAgentSetting(String key, String value) {
 
         // Core:
         // forbid 'enabled', must remain enabled to dynamically enable tracing / metrics
-        // forbid 'recording' / 'instrument', controlled by 'telemetry.metrics.enabled' / 'telemetry.tracing.enabled'
+        // forbid 'recording', controlled by 'telemetry.metrics.enabled' / 'telemetry.tracing.enabled'
+        "instrument",
```
Comment (mosche) [inline modules/apm/src/main/java/org/elasticsearch/telemetry/apm/internal/APMAgentSettings.java:122]: Whitelisting `instrument` so it can be enabled to investigate ES-7806

File: modules/apm/src/main/java/org/elasticsearch/telemetry/apm/internal/APMAgentSettings.java [hunk 2]
```diff
@@ -44,7 +44,6 @@ public void addClusterSettingsListeners(ClusterService clusterService, APMTeleme
 
         clusterSettings.addSettingsUpdateConsumer(TELEMETRY_TRACING_ENABLED_SETTING, enabled -> {
             apmTracer.setEnabled(enabled);
-            this.setAgentSetting("instrument", Boolean.toString(enabled));
```
Comment (mosche) [inline modules/apm/src/main/java/org/elasticsearch/telemetry/apm/internal/APMAgentSettings.java:47]: This was accidentally enabled in https://github.com/elastic/elasticsearch/pull/99832/files#diff-8e5ca1315006cc00c58ef42d4bfc14f50b30d11567a845f8e551ecf3e52c830bR58, though it's not required for recording manual traces.

File: build-tools/src/main/java/org/elasticsearch/gradle/testclusters/RunTask.java [hunk 3]
```diff
@@ -213,8 +213,8 @@ public void beforeStart() {
                 // if metrics were not enabled explicitly for gradlew run we should disable them
                 else if (node.getSettingKeys().contains("telemetry.metrics.enabled") == false) { // metrics
                     node.setting("telemetry.metrics.enabled", "false");
-                } else if (node.getSettingKeys().contains("telemetry.tracing.enabled") == false
-                    && node.getSettingKeys().contains("tracing.apm.enabled") == false) { // tracing
+                } else if ((node.getSettingKeys().contains("telemetry.tracing.enabled")
+                    || node.getSettingKeys().contains("tracing.apm.enabled")) == false) { // tracing
```
Comment (pgomulka) [inline build-tools/src/main/java/org/elasticsearch/gradle/testclusters/RunTask.java:217]: I would prefer we remove the old setting being used here.
we should try not have deprecated settings used in our own codebase

**Review by pgomulka — APPROVED** [review]
LGTM,
left suggestion for deprecated setting in RunTask

File: build-tools/src/main/java/org/elasticsearch/gradle/testclusters/RunTask.java (hunk 3 above)
Comment (mosche) [reply build-tools/src/main/java/org/elasticsearch/gradle/testclusters/RunTask.java]: Done 👍

File: modules/apm/src/main/java/org/elasticsearch/telemetry/apm/internal/APMAgentSettings.java (hunk 1 above)
Comment (pgomulka) [reply modules/apm/src/main/java/org/elasticsearch/telemetry/apm/internal/APMAgentSettings.java:122]: do we need to have this change for resolving a bug? If you need to investigate this, you can make a change locally, right?

Comment (mosche) [reply modules/apm/src/main/java/org/elasticsearch/telemetry/apm/internal/APMAgentSettings.java:122]: You certainly could do that locally as well 👍 
Question is, if we want to forbid enabling automatic instrumentation. What's your preference?

Comment (mosche) [reply modules/apm/src/main/java/org/elasticsearch/telemetry/apm/internal/APMAgentSettings.java:122]: Actually, thinking about it again, it's probably better to not allow it as long as it's not working properly

File: build-tools/src/main/java/org/elasticsearch/gradle/testclusters/RunTask.java [hunk 4]
```diff
@@ -213,10 +213,9 @@ public void beforeStart() {
                 // if metrics were not enabled explicitly for gradlew run we should disable them
                 else if (node.getSettingKeys().contains("telemetry.metrics.enabled") == false) { // metrics
                     node.setting("telemetry.metrics.enabled", "false");
-                } else if (node.getSettingKeys().contains("telemetry.tracing.enabled") == false
-                    && node.getSettingKeys().contains("tracing.apm.enabled") == false) { // tracing
-                        node.setting("telemetry.tracing.enable", "false");
-                    }
+                } else if (node.getSettingKeys().contains("telemetry.tracing.enabled") == false) { // tracing
+                    node.setting("telemetry.tracing.enable", "false");
```
Comment (stu-elastic) [inline build-tools/src/main/java/org/elasticsearch/gradle/testclusters/RunTask.java:217]: This change seems unrelated.

File: build-tools/src/main/java/org/elasticsearch/gradle/testclusters/RunTask.java (hunk 3 above)
Comment (stu-elastic) [reply build-tools/src/main/java/org/elasticsearch/gradle/testclusters/RunTask.java]: I see this was requested.  I suggest moving this to a different PR and including a test.  This PR should focus on removing instrument.

**Review by stu-elastic — APPROVED** [review]
lgtm, thanks for tracking this down.

File: build-tools/src/main/java/org/elasticsearch/gradle/testclusters/RunTask.java (hunk 3 above)
Comment (mosche) [reply build-tools/src/main/java/org/elasticsearch/gradle/testclusters/RunTask.java]: Moved to https://github.com/elastic/elasticsearch/pull/105076

**Comment by mark-vieira** [comment]
Thanks @mosche!
